libyul/backends/evm/ControlFlowGraphBuilder.cpp

```diff
@@ -248,9 +241,11 @@ void ControlFlowGraphBuilder::operator()(Block const& _block)
 
 void ControlFlowGraphBuilder::operator()(If const& _if)
 {
+	StackSlot condition = std::visit(*this, *_if.condition);
 	auto& ifBranch = m_graph.makeBlock(debugDataOf(_if.body));
+	// TODO What is the correct debug data to use here?
```

Comment:
@ekpyron can you answer this question?

Reply:
You mean for the ``afterIf`` block? That's returning to the scope containing the if, so I guess whatever ``debugDataOf(*m_currentBlock)`` *before* the ``if``, i.e. at the beginning of this function? But since that shouldn't change during the visit to the condition, it should be fine as is.

But yeah ``afterIf`` is conceptually just the continuation of the block of the containing scope and the debug data of that scope should be stored in the current node when entering the ``if``.

Reply:
Isn't the main change in this PR that `m_currentBlock` can actually change during the visit of the if condition? I'll just grab the debug data before the visit and then reuse it.

Reply:
I ended up just moving the `makeBlock` calls above the visit to the condition.